### PR TITLE
Add e2e test for config generation

### PR DESCRIPTION
**Pull request by smukherj1**
1. Add a Hello World example for Java
2. Add a Go binary that sets up a Bazel test repo by copying the existing Hello World C++ & Java examples and runs a Bazel remote build using the configs generated and uploaded by the tools added in #936 & #937

### CODE REVIEW

File: tests/scripts/configs_e2e/configs_e2e.go [hunk 1]
```diff
@@ -0,0 +1,408 @@
...
+		"examples/remotebuildexecution/hello_world/cc/say_hello.cc",
+		"examples/remotebuildexecution/hello_world/cc/say_hello.h",
+
+		// Java Hello World example.
+		"examples/remotebuildexecution/hello_world/java/BUILD",
+		"examples/remotebuildexecution/hello_world/java/HelloWorld.java",
+	}
+
+	// workspaceTemplate is the template to create the Bazel WORKSPACE file in the test repo.
+	workspaceTemplate = template.Must(template.New("WORSPACE").Parse(`
```
Comment (DaveGay) [inline tests/scripts/configs_e2e/configs_e2e.go:71]:
```suggestion
	workspaceTemplate = template.Must(template.New("WORKSPACE").Parse(`
```

File: tests/scripts/configs_e2e/configs_e2e.go [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+# correctness and repeatability of the build.
+build:remote --incompatible_strict_action_env=true
+
+build:remote --remote_timeout=3600
+
+# Enable authentication. This will pick up application default credentials by
+# default. You can use --google_credentials=some_file.json to use a service
+# account credential instead.
+build:remote --google_default_credentials=true
+`, rbeInst)
```
Comment (DaveGay) [inline tests/scripts/configs_e2e/configs_e2e.go:211]: Consider moving this line up to :188 and starting a third Fprintf.

File: tests/scripts/configs_e2e/configs_e2e.go [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+// of Bazel.
+func runTestBuild(ctx context.Context, workingDir, bazelVersion string) error {
+	bazeliskPath, err := downloadBazelisk(workingDir)
+	if err != nil {
+		return fmt.Errorf("failed to download Bazelisk: %w", err)
+	}
+	if err := os.Chmod(bazeliskPath, os.ModePerm); err != nil {
+		return fmt.Errorf("unable to update the permissions of downloaded Bazelisk binary %q to make it executable: %w", bazeliskPath, err)
+	}
+	ctxWithDeadline, cancel := context.WithTimeout(ctx, 3*time.Minute)
```
Comment (DaveGay) [inline tests/scripts/configs_e2e/configs_e2e.go:327]: Maybe pull the `3*time.Minute` bit out to a constant, or command line flag-backed var?

File: tests/scripts/configs_e2e/configs_e2e.go [hunk 4]
```diff
@@ -0,0 +1,408 @@
...
+	log.Printf("--dest_root=%q", *destRoot)
+	log.Printf("--rbe_instance=%q", *rbeInstance)
+
+	ctx := context.Background()
+
+	m, err := downloadManifest(*manifestURL)
+	if err != nil {
+		log.Fatalf("Unable to download the manifest from %q: %v", *manifestURL, err)
+	}
+	log.Printf("Successfully downloaded the JSON manifest from %s.", *manifestURL)
```
Comment (DaveGay) [inline tests/scripts/configs_e2e/configs_e2e.go:391]:
```suggestion
	log.Printf("Successfully downloaded the JSON manifest from %q.", *manifestURL)
```

Keeping the trailing . off of URLs helps prevent frustration.  Ditto :403, :405, :407.

**Review by DaveGay — APPROVED** [review]

File: tests/scripts/configs_e2e/configs_e2e.go (hunk 1 above)
Comment (smukherj1) [reply tests/scripts/configs_e2e/configs_e2e.go:71]: Done

File: tests/scripts/configs_e2e/configs_e2e.go (hunk 2 above)
Comment (smukherj1) [reply tests/scripts/configs_e2e/configs_e2e.go:211]: Done

File: tests/scripts/configs_e2e/configs_e2e.go (hunk 3 above)
Comment (smukherj1) [reply tests/scripts/configs_e2e/configs_e2e.go:327]: Done. Moved to flag

File: tests/scripts/configs_e2e/configs_e2e.go (hunk 4 above)
Comment (smukherj1) [reply tests/scripts/configs_e2e/configs_e2e.go:391]: Done

**Pull request merged**